### DISCUSSION TITLE
Question to unicode filenames

### DISCUSSION THREAD

**Discussion by herb52** [ImageMagick/ImageMagick#6008] [Development]
### ImageMagick version

7.1.10-58

### Operating system

Windows

### Operating system, version and so on

Windows 10

### Description

Hello,

On my Windows 10 system I use ImageMagick to convert images into *.jpg images.

It works without problem using filenames that contain only ASCII characters.
Also with chinese characters I have seen no problems.

But with a filename that contains an emoji (a unicode character that needs surrogates) it does not work.
Now my question is:  is it by design or is it an error?

In case of it is by design, what are the restrictions for unicode characters in a filename.

Thanks for your help in advance
Best regards
herb

### Steps to Reproduce

Use a file with an emoji in filename e.g.

magick.exe convert -quiet -strip -identify -clamp -depth 8 test😔😔.avif test😔😔.jpg

**Comment by urban-warrior** [comment]
ImageMagick does not support emoji's.    We'll add the need for support to the wish list.

**Comment by herb52** [comment]
Hello,

thanks for your answer.
For me it would be helpful, that ImageMagick will support ALL unicode characters.

Please do not misunderstand when I ask again: which unicode-characters does the current version support?

Thanks in advance
Best regards
herb

**Reply by urban-warrior** [reply]
ImageMagick supports UTF-8.  Emoji's apparently require special handling concerning Unicode surrogate pairs which we need to investigate and add support.

**Comment by snibgo** [comment]
ImageMagick can have emojis in filenames. For example, with IM v7.1.0-49 on Windows 11. Here is a copy-paste of the text of a command window:

```
C:\web\im\emoji>%IMG7%magick rose: 🤯.png

C:\web\im\emoji>dir /od *.png
 Volume in drive C is XXXXXXXXXXX
 Volume Serial Number is XXXXXXXXXX

 Directory of C:\web\im\emoji

29/01/2023  17:02             7,027 🤯.png
```
I suppose this is simply using UTF-8. Sadly, Microsoft tried to popularize UTF-16 which can only cope with 65536 codepoints, and emojis have codepoints greater than that, so someone invented "surrogate pairs", which is a pair of 16-bit unsigned integers. This is a kludge to fix a kludge.

The command window shows the emoji in colour. Here is a screen grab of that command window:

![c](https://user-images.githubusercontent.com/33812368/215343721-7633d0df-9ab3-4caa-81f1-832b630b905b.png)

Windows 11 seems to handle UTF-8 more easily than Windows 8.1 did.
